### PR TITLE
create tests fot registration service

### PR DESCRIPTION
**Pull request by karvatskyi**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/OwnRuntimeException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.service;
+
+public class OwnRuntimeException extends RuntimeException {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/OwnRuntimeException.java:3]: You took the assignment too literally :) Change the name to reflect its essence

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -5,10 +5,26 @@
...
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int DEFAULT_SIZE_OF_PASSWORD_AND_LOGIN = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null
+                || user.getLogin().length() < DEFAULT_SIZE_OF_PASSWORD_AND_LOGIN) {
+            throw new OwnRuntimeException("Login must be longer then 6 symbol");
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:15]: don't hardcode the length limit in the exception message, use your constant. Fix everywhere

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    @Test
+    void nullAge_NotOk() {
+        user.setAge(null);
+        assertThrows(OwnRuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void smallerThenEighteen_NotOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:47]:
```suggestion
    void underage_NotOk() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Test
+    void registeredUser_NotOk() {
+        boolean actual = storage.people.contains(user.getLogin());
+        assertFalse(actual);
+    }
+
+    @Test
+    void passwordSmallerThenSix_NotOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:61]:
```suggestion
    void passwordTooShort_NotOk() {
```
you may change the length limit at some point, and then have to deal with the hustle of renaming methods, Fix everywhere

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    private Storage storage;
+
+    @BeforeEach
+    public void setup() {
+        storage = new Storage();
+        user = new User();
+    }
+
+    @Test
+    void nullLogin_NotOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:23]: try to adhere to the naming convention mentioned in the description/checklist. 
image

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+    private Storage storage;
+
+    @BeforeEach
+    public void setup() {
+        storage = new Storage();
+        user = new User();
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:19]: user should be fully setup here. When you in your test set password to null, and expect your exception to be thrown, it's actually thrown because login is null, and login check goes first. Your test don't actually test what you think they test

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 7]
```diff
@@ -5,10 +5,26 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int DEFAULT_SIZE_OF_PASSWORD_AND_LOGIN = 6;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:8]: better separate into 2 constants, in case at some point in the future they no longer will be the same, it will be easier to update values

**Review by okuzan — CHANGES_REQUESTED** [review]
Change the spelling then to than where needed. Test edge cases too - age exactly 18, password/login - 6.

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Test
+    void register_anExisting_Login_Not_Ok() {
+        Storage.people.add(user);
+        try {
+            registrationService.register(user);
+        } catch (ValidationException e) {
+            return;
+        }
+        fail("Exception must be throw");
```
Comment (olekskov) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:44]: try/catch very rarely used and/or required in tests.
Please avoid it and use assertThrows structure for all tests

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java (hunk 6 above)
Comment (olekskov) [reply src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:21]: Oleh's comment not fixed, you always put new empty user to test and always get same exception.
You need to fill user with all proper fields in BeforeEach method and then for each test change required field to perform proper testing (login to null, password too short etc.)

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+    private StorageDao storageDao = new StorageDaoImpl();
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:17]:
```suggestion
    private static RegistrationService registrationService;
    private User user;
    private StorageDao storageDao;
```

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 10]
```diff
@@ -5,10 +5,35 @@
...
-        return null;
+        if (user == null) {
+            throw new ValidationException("User cannot be null");
+        }
+        final String login = user.getLogin();
+        final String password = user.getPassword();
+        final Integer age = user.getAge();
+
+        if (storageDao.get(login) != null) {
+            throw new ValidationException("Such user is register");
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:23]:
```suggestion
            throw new ValidationException("This user is already registered");
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        Storage.people.clear();
+        user = new User();
+        user.setLogin("Login123");
+        user.setPassword("password123");
+        user.setAge(24);
+    }
+
+    @Test
+    void register_validUser_registrationSuccessful() {
+        User registeredUser = registrationService.register(user);
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:32]: assert that the user is added to storage too

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job, small amends left

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 12]
```diff
@@ -5,10 +5,35 @@
...
+        if (password == null || password.length() < DEFAULT_PASSWORD_SIZE) {
+            throw new ValidationException("Password must be longer than "
+                    + DEFAULT_PASSWORD_SIZE + " characters");
+        }
+        if (age == null || age < MINIMUM_AGE_FOR_REGISTER) {
+            throw new ValidationException("You must be older than " + MINIMUM_AGE_FOR_REGISTER);
+        }
+        storageDao.add(user);
+        return user;
     }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:38]:
```suggestion
        return storageDao.add(user);
    }
```

**Review by aaananas — APPROVED** [review]
